Components/Queues/AsyncQueueConsumer.php

```diff
@@ -137,26 +128,39 @@ protected function process(EndpointInterface $queueEndpoint, MessageInterface $m
      * Overrides the main callback function to convert the AMQPMessage from the queue into a QueueMessage.
      *
      * {@inheritdoc}
-     *
-     * @return callable
      */
     protected function callback(EndpointInterface $endpoint): callable
     {
-        return function (AMQPMessage $message) use ($endpoint) {
+        return function (AMQPMessage $encodedMessage) use ($endpoint) {
+            $driver = $this->getQueueDriver($endpoint);
```

Comment:
Can't we move this assignment inside the `catch`? It's a call to the container that might end up not being used.

Reply:
Yes, definitely, it was there because before was also used on the `try` part but now is not anymore and I missed to move. Good catch

Reply:
Changed